connector/profiler/src/main/scala/org/apache/spark/executor/profiler/ExecutorJVMProfiler.scala

```diff
@@ -0,0 +1,171 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.executor.profiler
+
+import java.io.{BufferedInputStream, FileInputStream, InputStream}
+import java.net.URI
+import java.util.concurrent.{ScheduledExecutorService, TimeUnit}
+
+import one.profiler.{AsyncProfiler, AsyncProfilerLoader}
+import org.apache.hadoop.fs.{FileSystem, FSDataOutputStream, Path}
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.ThreadUtils
+
+
+/**
+ * A class that enables the async code profiler
+ */
+private[spark] class ExecutorJVMProfiler(conf: SparkConf, executorId: String) extends Logging {
+
+  private var running = false
+  private val enableProfiler = conf.get(EXECUTOR_CODE_PROFILING_ENABLED)
+  private val profilerOptions = conf.get(EXECUTOR_CODE_PROFILING_OPTIONS)
+  private val profilerOutputDir = conf.get(EXECUTOR_CODE_PROFILING_OUTPUT_DIR)
+  private val profilerLocalDir = conf.get(EXECUTOR_CODE_PROFILING_LOCAL_DIR)
+  private val writeInterval = conf.get(EXECUTOR_CODE_PROFILING_WRITE_INTERVAL)
+
+  private val startcmd = s"start,$profilerOptions,file=$profilerLocalDir/profile.jfr"
+  private val stopcmd = s"stop,$profilerOptions,file=$profilerLocalDir/profile.jfr"
+  private val dumpcmd = s"dump,$profilerOptions,file=$profilerLocalDir/profile.jfr"
+  private val resumecmd = s"resume,$profilerOptions,file=$profilerLocalDir/profile.jfr"
+
+  private val UPLOAD_SIZE = 8 * 1024 * 1024 // 8 MB
+  private var outputStream: FSDataOutputStream = _
+  private var inputStream: InputStream = _
+  private val dataBuffer = new Array[Byte](UPLOAD_SIZE)
+  private var threadpool: ScheduledExecutorService = _
+  private var writing: Boolean = false
+
+  val profiler: AsyncProfiler = if (enableProfiler) {
+    if (AsyncProfilerLoader.isSupported) {
+      AsyncProfilerLoader.load()
+    } else {
+      logWarning("Executor code profiling is enabled but is not supported for this platform")
+      null
+    }
+  } else {
+    null
+  }
+
+  def start(): Unit = {
+    if (profiler != null && !running) {
+      logInfo("Executor code profiling starting.")
+      try {
+        profiler.execute(startcmd)
+      } catch {
+        case e: Exception =>
+          logWarning("Executor code profiling aborted due to exception: ", e)
+          return
```

Comment:
May I ask why we prevent the exception propagation? I believe it's better instead of ignoring like this.